AdaptiveDynaMOSA implementation

giograno opened the pull request:
Hi all, this is the implementation of Adaptive DynaMOSA from the [paper](https://ieeexplore.ieee.org/document/8865437) "_Testing with Fewer Resources: An Adaptive Approach to Performance-Aware Test Case Generation_".

Probably @apanichella is one of the best candidates to look a it :)

apanichella commented:
I'll give a look over the weekend (and make some local tests). I'll keep you posted.

@giograno can you add some test cases?

giograno commented:
Thanks @apanichella! 

In the meanwhile, I added a few tests, especially for the sorting of the performance score and the indicators. 
Let me know if you need something else and/or you have any question.

apanichella commented:
Great! I am gonna run some experiments to be sure we don't break the latest changes in the main/master branch

giograno commented:
Hi there! Is there any news about this PR? Is still under consideration? Thanks :)

shabbirabdul commented on client/src/main/java/org/evosuite/ga/metaheuristics/mosa/AdaptiveDynaMOSA.java:
```diff
@@ -0,0 +1,252 @@
+//        logger.debug("N. fronts = {}", this.rankingFunction.getNumberOfSubfronts());
+//        logger.debug("1* front size = {}", this.rankingFunction.getSubfront(0).size());
+//        logger.debug("Best Values Map size = {}", goalsManager.getBestValues().size());
+        logger.debug("Covered goals = {}", goalsManager.getCoveredGoals().size());
+        logger.debug("Current goals = {}", goalsManager.getCurrentGoals().size());
+        logger.debug("Uncovered goals = {}", goalsManager.getUncoveredGoals().size());
```
Can be condensed to 

`
logger.debug("Covered goals = {}, Current goals = {} and Uncovered goals = {}", goalsManager.getCoveredGoals().size(),   goalsManager.getCurrentGoals().size(), goalsManager.getUncoveredGoals().size())
`

shabbirabdul commented on client/src/main/java/org/evosuite/ga/metaheuristics/mosa/AdaptiveDynaMOSA.java:
```diff
@@ -0,0 +1,252 @@
+        }
+
+        this.currentIteration++;
+//        logger.debug("N. fronts = {}", this.rankingFunction.getNumberOfSubfronts());
+//        logger.debug("1* front size = {}", this.rankingFunction.getSubfront(0).size());
+//        logger.debug("Best Values Map size = {}", goalsManager.getBestValues().size());
```
Commented code is harmless, however commented code greatly reduces the readability of code. I suggest we remove these commented lines.

giograno commented:
Hi @shabbirabdul, many thanks for your comments. I have addressed them! Plus, I rebased the implementation of AdaptiveDynaMOSA on the master, since a lot has changed from when I opened the PR.

giograno commented:
Hi there, I stumble across this PR, close to have its first birthday :) @apanichella do you have maybe any update? Many thanks 🙏

giograno commented:
Hey there. This PR grew quite old and it never seemed to be under consideration. I am going to close it. 
Please, feel free to re-open it in case you are interested to the contribution. Thank you very much.
